### PR TITLE
[DOC] Add more tests for global alignment.

### PR DESCRIPTION
**Pull request by rrahn**
* Adds more tests for the global alignment. These tests will be reused to test collections with parallel and vectorised execution.
* Adds tests for corner cases, e.g. aligning against empty sequences.
* Renamed the test fixtures to better compare them.
* Removed the global using declaration for seqan3 namespaces.

### CODE REVIEW

File: test/unit/alignment/pairwise/fixture/global_affine_unbanded.hpp [hunk 1]
```diff
@@ -135,4 +135,255 @@ static auto dna4_02 = []()
...
+        /*A*/  u  ,DuL,
+        /*G*/  u  ,uL ,
+        /*C*/  u  ,uL ,
+        /*G*/  u  ,uL
+        }
+    };
+}();
+
+
+static auto dna4_match_4_mismatch_5_gap_1_open_10_seq1_empty_part_06 = []()
```
Comment (eseiler) [inline test/unit/alignment/pairwise/fixture/global_affine_unbanded.hpp]: Do you want to have the `part_06` part here (and the next two)? The `seq_empty` makes it already different from the others before.

File: test/unit/alignment/pairwise/pairwise_alignment_single_test_template.hpp [hunk 2]
```diff
@@ -15,14 +15,12 @@
 
 #include "fixture/alignment_fixture.hpp"
 
-using namespace seqan3;
-using namespace seqan3::detail;
-using namespace seqan3::test::alignment::fixture;
+// using namespace seqan3::test::alignment::fixture;
```
Comment (eseiler) [inline test/unit/alignment/pairwise/pairwise_alignment_single_test_template.hpp]: remove

File: test/unit/alignment/pairwise/pairwise_alignment_single_test_template.hpp [hunk 3]
```diff
@@ -36,6 +34,8 @@ TYPED_TEST_CASE_P(pairwise_alignment_test);
 
 TYPED_TEST_P(pairwise_alignment_test, score)
 {
+    using namespace seqan3;
```
Comment (eseiler) [inline test/unit/alignment/pairwise/pairwise_alignment_single_test_template.hpp]: remove and do qualified calls

File: test/unit/alignment/pairwise/fixture/global_affine_unbanded.hpp (hunk 1 above)
Comment (rrahn) [reply test/unit/alignment/pairwise/fixture/global_affine_unbanded.hpp]: probably not. I did wanted to know that they are special because of empty sequences.

File: test/unit/alignment/pairwise/pairwise_alignment_single_test_template.hpp (hunk 3 above)
Comment (rrahn) [reply test/unit/alignment/pairwise/pairwise_alignment_single_test_template.hpp]: 😭

File: test/unit/alignment/pairwise/fixture/global_affine_unbanded.hpp (hunk 1 above)
Comment (eseiler) [reply test/unit/alignment/pairwise/fixture/global_affine_unbanded.hpp]: Do as you like

File: test/unit/alignment/pairwise/pairwise_alignment_single_test_template.hpp (hunk 3 above)
Comment (eseiler) [reply test/unit/alignment/pairwise/pairwise_alignment_single_test_template.hpp]: 😈

File: test/unit/alignment/pairwise/fixture/global_affine_unbanded.hpp [hunk 4]
```diff
@@ -24,15 +24,18 @@
...
+inline constexpr auto align_config_dna_score = align_config
+                                             | seqan3::align_cfg::scoring{
+                                                    seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
+                                                                                      seqan3::mismatch_score{-5}}};
 
-inline constexpr auto align_config = align_cfg::mode{global_alignment} |
-                                     align_cfg::gap{gap_scheme{gap_score{-1}, gap_open_score{-10}}};
-
-static auto dna4_01 = []()
+static auto dna4_match_4_mismatch_5_gap_1_open_10_part_01 = []()
```
Comment (smehringer) [inline test/unit/alignment/pairwise/fixture/global_affine_unbanded.hpp]: Question: why is it neccessary to include the paramters in the name? It feels like this could lead to inconsistencies in the future if you change some parameters (especially since they are at the top not inside of the lambda)

EDIT: also if you are this expressive it is weird that part 01-05 are not expressive in a sense that I do not now what is testing what? I see that all tests here have the same scoring scheme anyway. It feels more helpful if the names would be `dna4_default_scoring_scheme_score_only` and then rename your `align_config_dna_score` to `align_config_dna_default_score` or something similar.

File: test/unit/alignment/pairwise/fixture/global_affine_unbanded.hpp [hunk 5]
```diff
@@ -135,4 +139,261 @@ static auto dna4_02 = []()
...
+        /*C*/  u  ,uL ,
+        /*T*/  u  ,uL ,
+        /*A*/  u  ,DuL,
+        /*G*/  u  ,uL ,
+        /*C*/  u  ,uL ,
+        /*G*/  u  ,uL
+        }
+    };
+}();
+
```
Comment (smehringer) [inline test/unit/alignment/pairwise/fixture/global_affine_unbanded.hpp]: remove extra newline :nail_care:

File: test/unit/alignment/pairwise/fixture/global_affine_unbanded.hpp (hunk 4 above)
Comment (rrahn) [reply test/unit/alignment/pairwise/fixture/global_affine_unbanded.hpp]: Because I want to identify tests with the same configurations, which is necessary for testing vectorised alignments. 
So there might be added tests with different scores that cannot be put into the same vectorisation code - part01 - 05 indicates that they have the same configuration but different sequences. Not sure why this would be an issue?
